craft_grammar/errors.py

```diff
@@ -24,14 +24,28 @@ class CraftGrammarError(Exception):
 class PlatformNameError(CraftGrammarError):
     """Error raised if a platform name is invalid."""
 
-    def __init__(self, *platforms: str) -> None:
-        if len(platforms) == 1:
-            super().__init__(f"Invalid platform name: '{platforms[0]}'")
+    def __init__(self, *platform: str) -> None:
+        if len(platform) == 1:
+            super().__init__(f"Invalid platform name: '{platform[0]}'")
         else:
-            platforms_str = "', '".join(platforms)
+            platforms_str = "', '".join(platform)
             super().__init__(f"Invalid platform names: '{platforms_str}'")
 
 
+class UnknownPlatformNameError(CraftGrammarError):
+    """Error raised if for statement contains an unknown platform name."""
+
+    def __init__(self, platform: str) -> None:
+        super().__init__(f"Undefined platform name: {platform!r}")
```

Comment:
```suggestion
        super().__init__(f"Unknown platform name: {platform!r}")
```

Out of coherence with `UnknownArchitectureError`